server/views/applications/view.njk

```diff
@@ -15,95 +15,30 @@
+        <dd class="govuk-summary-list__value">{{ this_data.app.terms_and_conditions_url }}</dd>
         <dd class="govuk-summary-list__actions"></dd>
       </div>
     </dl>
-    <a href="/manage-applications/1/update" class="govuk-button govuk-button--secondary" role="button" draggable="false">Manage application</a>
+    <a href="/manage-applications/{{ this_data.app.id }}/update/{{ this_data.env }}" class="govuk-button govuk-button--secondary" role="button" draggable="false">Manage application</a>
     <h2 class="govuk-heading-l">Keys for this application</h2>
-    <dl class="govuk-summary-list govuk-body govuk-!-font-size-16">
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          Key name
-        </dt>
-        <dd class="govuk-summary-list__value">REST API Key for testing</dd>
-        <dd class="govuk-summary-list__actions">
-          <p class="govuk-body govuk-!-font-size-16">
-            <a href="/manage-applications/{{ this_data.appId }}/api-key/update">Update key</a>
-          </p>
-          <p class="govuk-body govuk-!-font-size-16 govuk-!-margin-bottom-0">
-            <a href="/manage-applications/:appId/api-key/delete">Delete key</a>
-          </p>
-        </dd>
-      </div>
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          Key type
-        </dt>
-        <dd class="govuk-summary-list__value">
-          Rest API key
-        </dd>
-        <dd class="govuk-summary-list__actions"></dd>
-      </div>
-
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          API key ID
-        </dt>
-        <dd class="govuk-summary-list__value">
-              <span class="api-value">
-                ff78ff7b-542c-4a2c-85ba-c8ebe26ad23f</span>
-        </dd>
-        <dd class="govuk-summary-list__actions">
-          <a class="govuk-link" href="#">
-            Copy<span class="govuk-visually-hidden">
-                  key</span>
-          </a>
-        </dd>
-      </div>
-
-
-
-
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          Restricted IPs
-        </dt>
-        <dd class="govuk-summary-list__value">
-          194.75.36.81
-        </dd>
-        <dd class="govuk-summary-list__actions"></dd>
-      </div>
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          JavaScript domains
-        </dt>
-        <dd class="govuk-summary-list__value"></dd>
-        <dd class="govuk-summary-list__actions"></dd>
-      </div>
-      <div class="govuk-summary-list__row">
-        <dt class="govuk-summary-list__key">
-          Redirect URIs
-        </dt>
-        <dd class="govuk-summary-list__value"></dd>
-        <dd class="govuk-summary-list__actions"></dd>
-      </div>
-    </dl>
+      {% for key in this_data.keys.items %}
+        {% include "_partials/key_detail.njk" %}
```

Comment:
- Seeing as `key_detail.njk` is a partial that is not used anywhere else but here, and only relevant to the application route, it should be moved to  the `view/application` folder.. The `_partials` folder is only for html fragments re-used across multiple views or layouts.
- Also, because it does not match directly to a route handler, I'd prepend it with an underscore.
- In fact, if it's only ever needed in `views/application/view.njk`, then probably didn't need to be factored out at all 😄

Reply:
- Sorry didn't realise that was the convention we were following,

- Again, my mistake.

- I find it more readable and easy to follow and it restricts the code that needs to be be altered and re-reviewed if for whatever reason we need to that code. Can I get more opinions from the team?

Reply:
- Not sure what IDE you use but most IDEs I know make it quite easy navigating through a file of 116 lines (which is the combined size of `key_detail.njk` and `view.njk`).. 
- Also, no separation is necessary if `key_detail.njk` is not reused anywhere.